### PR TITLE
editoast: fix train arrival_time when adding a margin

### LINKED ISSUE #6695: Train schedule travel time isn't updated in timetable when adding margin
### What happened?

In a scenario, when adding margin to a train, the travel time of this train and the time arrival in the timetable aren't updated accordingly.

### What did you expect to happen?

The travel time and arrival time should be updated when adding a margin.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Toggle the timetable trains details by clicking on the eye icon at the top and check the travel time
3. Modify a train and add some margins
4. Check again the travel time and arrival time of the train, it hasn't changed

### What operating system, browser and environment are you using?

- **Browser**: Firefox v123
- **OS**: macOS Sonoma
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev f7339f1

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #6695

### CODE REVIEW

**Review by woshilapin — APPROVED** [review]
Nice bug fix. Do you think it's possible to add test on this, on order to avoid future regression?

File: editoast/src/models/timetable.rs [hunk 1]
```diff
@@ -125,6 +125,14 @@ impl Timetable {
...
                     eco,
                 };
+                let arrival_time = result_train_eco.as_ref().map_or(arrival_time, |eco| {
+                    eco.0
+                        .head_positions
+                        .last()
+                        .expect("Train should have at least one position")
+                        .time
+                        + train_schedule.departure_time
+                });
```
Comment (flomonster) [inline editoast/src/models/timetable.rs]: We can combine the two variable assignments into one.
There is no reason to define a first value on line 115 and replace it on line 128.

```suggestion
                let arrival_time = if let Some(eco) = result_train_eco {
                    eco.0.head_positions.last()
                } else {
                    result_train.head_positions.last()
                }
                .expect("Train should have at least one position")
                .time
                    + train_schedule.departure_time;
```

**Review by flomonster — APPROVED** [review]
The fix is fine to me. I just made a small suggestion

**Review by clarani — APPROVED** [review]
:white_check_mark: LGTM

**Pull request merged**